src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int GROWTH_FACTOR = 2;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size = 0;
+    private T[] elements = (T[]) new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growArrayIfFull();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexForAddition(index);
+        growArrayIfFull();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T temp = elements[index];
+
+        if (index != size - 1) {
+            System.arraycopy(elements, index + 1, elements, index, size - index);
+        }
```

Comment:
if is not needed

Reply:
Can you explain, why?
Because tests didn`t pass without this check

Reply:
![image](https://github.com/user-attachments/assets/2d534295-6f92-4b72-aa59-ecfa91a8fd02)